src/improvements/tasks/MultisigTask.sol

```diff
@@ -392,34 +414,72 @@ abstract contract MultisigTask is Test, Script {
+        execTransaction(_childMultisig, MULTICALL3_ADDRESS, 0, approveCalldata, Enum.Operation.DelegateCall, signatures);
     }
 
     /// @notice Executes the task with the given signatures.
     /// @param signatures The signatures to execute the task.
-    function execute(bytes memory signatures) public {
+    function execute(bytes memory signatures) public returns (VmSafe.AccountAccess[] memory) {
         bytes memory data = getCalldata();
         bytes32 hash = getHash();
 
-        signatures = Signatures.prepareSignatures(parentMultisig, hash, signatures);
+        if (signatures.length == 0) {
+            // if no signatures are attached, this means we are dealing with a
+            // nested safe that should already have all of its approve hashes in
+            // child multisigs signed already.
+            signatures = prepareSignatures(parentMultisig, hash);
+        } else {
+            // otherwise, if signatures are attached, this means EOA's have
+            // signed, so we order the signatures based on how Gnosis Safe
+            // expects signatures to be ordered by address cast to a number
+            signatures = Signatures.sortUniqueSignatures(
+                parentMultisig, signatures, hash, IGnosisSafe(parentMultisig).getThreshold(), signatures.length
+            );
+        }
+
+        vm.startStateDiffRecording();
+
+        execTransaction(parentMultisig, multicallTarget, 0, data, Enum.Operation.DelegateCall, signatures);
+
+        return vm.stopAndReturnStateDiff();
+    }
+
+    /// @notice helper function that returns whether or not the current context
+    /// is a broadcast context
+    function isBroadcastContext() internal view returns (bool) {
+        return vm.isContext(VmSafe.ForgeContext.ScriptBroadcast) || vm.isContext(VmSafe.ForgeContext.ScriptResume);
+    }
+
+    /// @notice executes a transaction to the target multisig
+    /// @param multisig to execute the transaction from
+    /// @param target to call when executing the transaction
+    /// @param value amount of value to send from the safe
+    /// @param data calldata to send from the safe
+    /// @param operationType type of operation to execute
+    /// @param signatures for the safe transaction
+    function execTransaction(
+        address multisig,
+        address target,
+        uint256 value,
+        bytes memory data,
+        Enum.Operation operationType,
+        bytes memory signatures
+    ) internal {
+        if (isBroadcastContext()) {
+            vm.broadcast();
+        }
+
+        bool success = false;
+
+        try IGnosisSafe(multisig).execTransaction(
+            target, value, data, operationType, 0, 0, 0, address(0), payable(address(0)), signatures
+        ) returns (bool execStatus) {
+            success = execStatus;
+        } catch (bytes memory err) {
+            console.log("Error executing multisig transaction");
+            console.logBytes(err);
+        }
```

Comment:
I'm pretty sure this extra `catch` block can never be entered because this is a high level call that only returns a bool, right? So we should switch back to the original approach you had, sorry. See https://github.com/ethereum-optimism/superchain-ops/pull/622#discussion_r1961775502

Reply:
Going to resolve this comment. Based on Elliots testing above.

Reply:
got it, that makes sense. thanks!